nextstrain/cli/command/init_shell.py

```diff
@@ -0,0 +1,102 @@
+"""
+Prints the shell init script for a Nextstrain CLI standalone installation.
+
+If PATH does not contain the expected installation path, emits an appropriate
+``export PATH=…`` statement.  Otherwise, emits only a comment.
+
+Use this command in your shell config with a line like the following::
+
+    source <({INSTALLATION_PATH}/nextstrain init-shell)
+
+Exits with error if run in an non-standalone installation.
+"""
+
+import os
+import shutil
+from pathlib import Path
+from shlex import quote as shquote
+from textwrap import dedent
+from typing import Optional
+from ..errors import UserError
+from ..util import standalone_installation_path
+
+
+try:
+    INSTALLATION_PATH = standalone_installation_path()
+except:
+    INSTALLATION_PATH = None
+
+__doc__ = (__doc__ or "").format(
+    INSTALLATION_PATH = (
+             shquote(str(INSTALLATION_PATH))
+          if INSTALLATION_PATH
+        else "…/path/to"
+    )
+)
```

Comment:
Curious, why `__doc__ or ""` instead of just `__doc__`?

Reply:
It's a minor wart solely to satisfy type checking since pyright thinks `__doc__` might be `None`.

```shell-session
$ cat /tmp/doctype.py 
"""
Hello, docstring!
"""
reveal_type(__doc__)

$ mypy /tmp/doctype.py 
/tmp/doctype.py:4: note: Revealed type is "builtins.str"
Success: no issues found in 1 source file

$ pyright /tmp/doctype.py 
Loading configuration file at /home/tom/nextstrain/cli/pyrightconfig.json
Assuming Python platform Linux
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
stubPath /home/tom/nextstrain/cli/typings is not a valid directory.
Searching for source files
Found 1 source file
pyright 1.1.270
/tmp/doctype.py
  /tmp/doctype.py:4:13 - information: Type of "__doc__" is "str | None"
0 errors, 0 warnings, 1 information 
Completed in 0.599sec
```

Reply:
Ohh I see. I guess it'd be clear when CI fails, but a comment wouldn't hurt. This sounds like a bad discrepancy between mypy and pyright. Not sure which one is actually correct, they're both hardcoded...

```
% cat no_docstring.py 
reveal_type(__doc__)

% mypy no_docstring.py 
no_docstring.py:1: note: Revealed type is "builtins.str"

% npx pyright no_docstring.py
...
  /tmp/no_docstring.py:1:13 - information: Type of "__doc__" is "str | None"
0 errors, 0 warnings, 1 information 
```

Reply:
I'll add a comment.

Some context on the pyright and mypy behaviour here:

- https://github.com/microsoft/pyright/issues/2241
- https://github.com/python/mypy/issues/9170
- https://github.com/python/mypy/issues/2709

from which it sounds like recent mypy's should be typing `__doc__` equivalently as `Optional[str]`, so not sure why we're not observing that (using most recent version on PyPI).

Reply:
Ok, the different is the `strict_optional = False` in our `mypy.ini`. Running mypy outside of the repo with `--no-strict-optional` produces the same result.

Reply:
https://github.com/python/mypy/blob/7c14feedd2a6889d9eab8b0ac8dc8aab630bbed3/mypy/nodes.py#L119-L126

https://github.com/python/mypy/blob/7c14feedd2a6889d9eab8b0ac8dc8aab630bbed3/mypy/semanal.py#L595-L599

Reply:
The Python data model [is pretty clear](https://docs.python.org/3/reference/datamodel.html?highlight=__doc__#the-standard-type-hierarchy) that a module's `__doc__` may be `None`.

![image](https://user-images.githubusercontent.com/79913/190705343-374c6a0a-0d00-4a84-b10f-a6fa42ed9984.png)

Reply:
Comment added in f6e2c8a.

Reply:
Haha, today's Python learning! Thanks for digging and adding the comment.